Drop support for Plone 5.2 and Plone 6.0

ericof opened the pull request:
- [x] Drop support for Plone 5.2
- [x] Drop support for Plone 6.0
- [x] Use plone/meta
- [x] Move tests to pytest
- [x] [Add `default` distribution](https://github.com/plone/plone.distribution/issues/80)
- [x] Resolve conflicts
- [x] Make sure tests run on Plone 6.1
- [x] Fix failing CI
- [x] Create default content

Fixes #155 

Before merge:
- [x] Create 4.x.x branch of plone.volto, update buildout.coredev 6.0 to use it
- [ ] Update required checks

mister-roboto commented:
@ericof thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

mauritsvanrees reviewed (commented):
How is anyone supposed to review this?
Can't you base this on my PR #153?  I split that up into multiple commits to make it easier to review: reviewers can ignore big commits like "black", "pyupgrade", or "zpretty", and focus on the much smaller real changes.
With some git magic it would probably still be possibly to do that, without getting dozens of merge conflicts.  (I don't much mind having wasted some time on my PR, that just sometimes happens when multiple people work on one project, and is not the real issue.)

Having said that: happy to see support for the older Plone versions dropped.
What is the reason for changing the i18n domain to `plone.volto`?

Oh, you may want to rename the news snippet from `155.internal` to `155.breaking`.  This is not quite an internal-only change. ;-)

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+</div>
+
+**plone.volto** configures [Plone](https://plone.org/) to work with [Volto](https://www.npmjs.com/package/@plone/volto), the default frontend Plone since version 6.0.
```
```suggestion
`plone.volto` configures [Plone](https://plone.org/) to work with [Volto](https://www.npmjs.com/package/@plone/volto), the default frontend of Plone since version 6.0.
```

stevepiercy commented on README.md:
````diff
@@ -0,0 +1,273 @@
+
+Add **plone.volto** to the Plone installation using `pip`:
+
+```bash
````
````suggestion
```shell
````

stevepiercy commented on README.md:
````diff
@@ -0,0 +1,273 @@
+pip install plone.volto
+```
+
+or add it as a dependency on your package's `setup.py`
````
```suggestion
or add it as a dependency in your package's `setup.py`:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+## Installation
+
+Add **plone.volto** to the Plone installation using `pip`:
```
```suggestion
Add `plone.volto` either to the Plone installation using `pip`:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+## Compatibility
+
+**plone.volto** version 5.x works with Plone 6.1 (pre-alpha and alpha), and supports Python 3.10, 3.11 and 3.12.
```
```suggestion
`plone.volto` version 5.x works with Plone 6.1 (pre-alpha and alpha), and supports Python 3.10, 3.11, and 3.12.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+**plone.volto** version 5.x works with Plone 6.1 (pre-alpha and alpha), and supports Python 3.10, 3.11 and 3.12.
+
+Though, Volto requires specific versions of plone.volto and plone.restapi:
```
```suggestion
Volto requires specific versions of `plone.volto` and `plone.restapi`:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+|  2.x (kitconcept.volto) | 5.1, 5.2 |  7.0.0 and above | New image scales |
+|  1.x (kitconcept.volto) | 5.1, 5.2 |  6.0.0 and below | New transforms and features |
+
+Volto only supports the latest plone.restapi branch, therefore it is recommended to always use the latest version in your Volto projects.
```
```suggestion
Volto only supports the latest `plone.restapi` branch, therefore it is recommended to always use the latest version in your Volto projects.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+| --- | --- | --- | --- |
+|  5.x | 6.1 |  8.41.0 and above | |
+|  4.x | 5.2, 6.0 |  8.41.0 and above | |
+|  3.x | 5.1, 5.2 |  8.13.0 and above | Requires new JSONSummarySerializerMetadata serializer added in plone.restapi 8.13.|
```
```suggestion
|  3.x | 5.1, 5.2 |  8.13.0 and above | Requires new `JSONSummarySerializerMetadata` serializer added in `plone.restapi` 8.13.|
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+## Plone 6 architecture
+
+Architectural Diagram of Plone 6::
```
```suggestion
Architectural diagram of Plone 6:
```

stevepiercy commented on README.md:
````diff
@@ -0,0 +1,273 @@
+
+Architectural Diagram of Plone 6::
+
+```
````
````suggestion
```text
````

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+## Features
+
+plone.volto provides the following features:
```
```suggestion
`plone.volto` provides the following features:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### Volto blocks support
+
+
+plone.volto enables the new Volto Blocks editor on `Document`, `Language Root Folder` and `Site Root`.
```
```suggestion
`plone.volto` enables the new Volto blocks editor on `Document`, `Language Root Folder`, and `Site Root`.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Block types index
+
+plone.volto adds a `block_types` index to the Plone catalog.
```
```suggestion
`plone.volto` adds a `block_types` index to the Plone catalog.
```

stevepiercy commented on README.md:
````diff
@@ -0,0 +1,273 @@
+portal_catalog.searchResults(block_types="image")
+```
+
+The `block_types` index was added in plone.volto 4.1.0.
````
```suggestion
The `block_types` index was added in `plone.volto` 4.1.0.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Multilingual support
+
+plone.volto supports multilingual websites.
```
```suggestion
`plone.volto` supports multilingual websites.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### Multilingual support
+
+plone.volto supports multilingual websites.
+Install PAM before installing this package and demo homepages will be created in each enabled language.
```
```suggestion
Install PAM before installing this package, and demo homepages will be created in each enabled language.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+plone.volto supports multilingual websites.
+Install PAM before installing this package and demo homepages will be created in each enabled language.
+Currently only support for EN/DE.
```
```suggestion
Support is currently only for `EN` and `DE`.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Document content type
+
+plone.volto disables the `Richtext` and `Table of Contents` behaviors for the `Document` content type.
```
```suggestion
`plone.volto` disables the `Richtext` and `Table of Contents` behaviors for the `Document` content type.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### Document content type
+
+plone.volto disables the `Richtext` and `Table of Contents` behaviors for the `Document` content type.
+Rich Text functionality is provided by the new Volto Blocks editor.
```
```suggestion
Rich text functionality is provided by the new Volto blocks editor.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+plone.volto disables the `Richtext` and `Table of Contents` behaviors for the `Document` content type.
+Rich Text functionality is provided by the new Volto Blocks editor.
+The `Table of Contents` functionality is provided by the `Table of Contents Block` in Volto.
```
```suggestion
The `Table of Contents` functionality is provided by the `Table of Contents` block in Volto.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### CORS profile
+
+A quick helper for enable CORS for development config is also provided in the
```
```suggestion
A quick helper to enable CORS for development configuration is also provided in the
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### CORS profile
+
+A quick helper for enable CORS for development config is also provided in the
+`plone.volto` module. So you can call:
```
```suggestion
`plone.volto` module. You can call:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### ZLog patch
+
+p.restapi low level errors are routed through the ancient ZLog and are `plone_error`
```
```suggestion
`plone.restapi` low level errors are routed through the ancient ZLog and are `plone_error`
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### ZLog patch
+
+p.restapi low level errors are routed through the ancient ZLog and are `plone_error`
+enabled, making it difficult to follow since all are marked with a UUID. Specially if
```
```suggestion
enabled, making it difficult to follow since all are marked with a UUID, specifically when
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Patch for `subject` field
+
+There are some problems of serialization on special characters derivated from how the
```
```suggestion
There are some problems of serialization on special characters derived from the
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+### Patch for `subject` field
+
+There are some problems of serialization on special characters derivated from how the
+current shape of the Plone's default Dexterity `subjects` field that has to be
```
```suggestion
current shape of the Plone's default Dexterity `subjects` field that have to be
```

stevepiercy commented on README.md:
````diff
@@ -0,0 +1,273 @@
+```
+
+There is also another variation of the preview image behavior called `volto.preview_image_link`.
+This one stores preview images using a relation to an Image content type, rather than in an image field. This might be preferable if many content items use the same preview image.
````
```suggestion
This one stores preview images using a relation to an image content type, rather than in an image field. This might be preferable if many content items use the same preview image.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Navigation title behavior
+
+The navigation title makes content types provide a nav_title field that is used by Volto in the main navigation, the breadcrumbs and the navigation portlet.
```
```suggestion
The navigation title makes content types provide a `nav_title` field that is used by Volto in the main navigation, the breadcrumbs, and the navigation portlet.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+The navigation title makes content types provide a nav_title field that is used by Volto in the main navigation, the breadcrumbs and the navigation portlet.
+
+The "volto.navtitle behavior can be enabled in the generic setup XML definition of a content type (e.g. "/profiles/default/types/MyContentType.xml"):
```
```suggestion
The `volto.navtitle` behavior can be enabled in the generic setup XML definition of a content type, for example in `/profiles/default/types/MyContentType.xml`:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Head title behavior
+
+The headtitle makes content types provide a headtitle field that can be used by Volto in teasers and alike.
```
```suggestion
The `headtitle` behavior makes content types provide a `headtitle` field that can be used by Volto in teasers and other content types.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+The headtitle makes content types provide a headtitle field that can be used by Volto in teasers and alike.
+
+The "volto.head_title" behavior can be enabled in the generic setup XML definition of a content type (e.g. "/profiles/default/types/MyContentType.xml"):
```
```suggestion
The `volto.head_title` behavior can be enabled in the generic setup XML definition of a content type, for example in `/profiles/default/types/MyContentType.xml`:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+### Image scales
+
+This package introduces new Plone image scales in Plone and redefines a couple of existing ones. These are know to work well with Volto layout and grid system:
```
```suggestion
This package introduces new Plone image scales in Plone and redefines a couple of existing ones.
These are know to work well with the Volto layout and grid system:
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+    </a>
+</p>
+
+The code of plone.volto has been under active development and is used in production since 2018.
```
```suggestion
The code of `plone.volto` has been under active development and is used in production since 2018.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+</p>
+
+The code of plone.volto has been under active development and is used in production since 2018.
+First as kitconcept.voltodemo (deprecated since March, 5th 2020), then as kitconcept.volto.
```
```suggestion
It was first named `kitconcept.voltodemo` (deprecated since March, 5th 2020), then `kitconcept.volto`.
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+
+The code of plone.volto has been under active development and is used in production since 2018.
+First as kitconcept.voltodemo (deprecated since March, 5th 2020), then as kitconcept.volto.
+In September 2021 kitconcept.volto has been renamed to plone.volto and has been contributed to the Plone core as part of [PLIP #2703](https://github.com/plone/Products.CMFPlone/issues/2703).
```
```suggestion
In September 2021 `kitconcept.volto` was renamed to `plone.volto`, and was contributed to the Plone core as part of [PLIP #2703](https://github.com/plone/Products.CMFPlone/issues/2703).
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+|  5.x | 6.1 |  8.41.0 and above | |
+|  4.x | 5.2, 6.0 |  8.41.0 and above | |
+|  3.x | 5.1, 5.2 |  8.13.0 and above | Requires new JSONSummarySerializerMetadata serializer added in plone.restapi 8.13.|
+|  2.x (kitconcept.volto) | 5.1, 5.2 |  7.0.0 and above | New image scales |
```
```suggestion
|  2.x (`kitconcept.volto`) | 5.1, 5.2 |  7.0.0 and above | New image scales |
```

stevepiercy commented on README.md:
```diff
@@ -0,0 +1,273 @@
+|  4.x | 5.2, 6.0 |  8.41.0 and above | |
+|  3.x | 5.1, 5.2 |  8.13.0 and above | Requires new JSONSummarySerializerMetadata serializer added in plone.restapi 8.13.|
+|  2.x (kitconcept.volto) | 5.1, 5.2 |  7.0.0 and above | New image scales |
+|  1.x (kitconcept.volto) | 5.1, 5.2 |  6.0.0 and below | New transforms and features |
```
```suggestion
|  1.x (`kitconcept.volto`) | 5.1, 5.2 |  6.0.0 and below | New transforms and features |
```

stevepiercy reviewed (changes requested):
I reviewed only `README.md`.

Can some of this content by moved into the open PR https://github.com/plone/documentation/pull/1671 ?

ericof commented:
I still need to fix dependencies and find out why tests fail in ci (but they pass locally)

ericof commented:
> How is anyone supposed to review this? Can't you base this on my PR #153? I split that up into multiple commits to make it easier to review: reviewers can ignore big commits like "black", "pyupgrade", or "zpretty", and focus on the much smaller real changes. With some git magic it would probably still be possibly to do that, without getting dozens of merge conflicts. (I don't much mind having wasted some time on my PR, that just sometimes happens when multiple people work on one project, and is not the real issue.)

I will make many PR's on top of yours. That way we can review one by one.

> 
> Having said that: happy to see support for the older Plone versions dropped. What is the reason for changing the i18n domain to `plone.volto`?

I changed it for 'laziness', but I see a use case for it now. Either way, I will roll it back to plone .

> 
> Oh, you may want to rename the news snippet from `155.internal` to `155.breaking`. This is not quite an internal-only change. ;-)
Thank you for pointing the obvious thing 😅

davisagli commented on setup.py:
```diff
@@ -54,26 +38,37 @@ def readfile(name):
         "plone.restapi>=8.41.0",
         "plone.app.vocabularies>=4.3.0",
+        "plone.distribution",
+        "Products.CMFPlacefulWorkflow",
```
Why does it depend on CMFPlacefulWorkflow?

ericof replied:
I do not remember anymore, but without this, package was breaking

davisagli commented on src/plone/volto/distributions/default/content/content/plone_site_root/data.json:
```diff
@@ -0,0 +1,37 @@
+  "@type": "Plone Site",
+  "UID": "plone_site_root",
+  "allow_discussion": null,
+  "blocks": {},
```
Shouldn't there be some default blocks?

ericof replied:
Yes, we need to have the default content here.

davisagli commented on src/plone/volto/distributions/default/schema.json:
```diff
@@ -0,0 +1,45 @@
+      },
+      "default_language": {"$ref": "#/definitions/languages"},
+      "portal_timezone": {"$ref": "#/definitions/timezones"},
+      "setup_content": {
```
Is this used?

ericof replied:
Yes, it is.

davisagli commented on src/plone/volto/patches.py:
```diff
@@ -16,14 +16,10 @@
 
-
-try:
-    from collective.folderishtypes.dx import content  # noqa F401
```
This breaks compatibility with collective.folderishtypes, if anyone is still using it, and needs to be mentioned as a breaking change in the release notes.

davisagli commented on src/plone/volto/profiles.zcml:
```diff
@@ -24,42 +24,6 @@
-      provides="Products.GenericSetup.interfaces.EXTENSION"
-      directory="profiles/demo"
-      post_handler=".setuphandlers.create_demo_homepage"
-      />
```
Removing these profiles should be mentioned as a breaking change in the release notes.

davisagli commented on news/155.internal:
We should also mention that plone.volto now includes the "default" distribution

davisagli commented on src/plone/volto/testing.py:
```diff
@@ -1,111 +1,121 @@
+    defaultBases = (BASE_FIXTURE,)
 
 
-PLONE_VOLTO_CORE_FIXTURE = PloneVoltoCoreLayer()
```
Please don't rename these fixtures. It's a breaking change for any package that uses them.

tisto commented:
The reason we are dropping support for Plone 5.2/6.0 is Plone Distributions, right?

davisagli commented:
> The reason we are dropping support for Plone 5.2/6.0 is Plone Distributions, right?

@tisto Right. plone.volto becomes the package which holds the default distribution. We'll create a 4.x branch where we can backport anything needed to continue supporting 5.2 and 6.0.

davisagli commented:
Testing on jenkins with this set of PRs:
```
https://github.com/plone/Products.CMFPlone/pull/3981
https://github.com/plone/plone.distribution/pull/90
https://github.com/plone/plone.volto/pull/156
https://github.com/plone/plone.classicui/pull/13
```

sneridagh commented:
> The following GenericSetup profiles were removed: `default-homepage`,
`default-homepage-drafjs`, `default-homepage-slate`, `demo` and `richtext`. @davisagli

in favor of what? Can I load a distribution in the robot server?

stevepiercy commented:
> @ericof @mauritsvanrees @davisagli @stevepiercy where has been documented the changes pushed by this PR? I cannot see it in: https://6.docs.plone.org/backend/upgrading/version-specific-migration/upgrade-to-61.html Trying to update Volto with it, and the profiles are gone. Where did they go? This has to be documented.

No one replied to my comment https://github.com/plone/plone.volto/pull/156#pullrequestreview-2143175238, so I took no action. I haven't been following the details of the changes here, so someone else needs to write it up, if they're needed.

sneridagh commented:
@ericof @davisagli I thought that removing the `plone.volto:default-homepage` profile would be enough, but nope.
The documents are not folderish and all the tests are broken. 
Please let me know what I have to do for the plone/server-acceptance image to work now.

mauritsvanrees commented:
@sneridagh Maybe add `plone.volto:default` instead?  Originally the `default-homepage` profile installed that one, with the only extra that it called [`create_default_homepage`](https://github.com/plone/plone.volto/blob/4.x.x/src/plone/volto/setuphandlers.py#L231).  So there is only a minor (though maybe important) difference.

Otherwise, the [`coresandbox` profile](https://github.com/plone/plone.volto/blob/5.0.2/src/plone/volto/profiles.zcml#L27-L35) is still available, but I don't know if that helps.
